getnacelle/nacelle-js
# feat: node 18

## Why are these changes introduced?

Addresses [ENG-9038](https://nacelle.atlassian.net/browse/ENG-9038) 

> Update nacelle-js to use Node 18

## What is this pull request doing?

This PR updates CI/CD to use Node.js 18 instead of Node.js 16 for task running. It also updates `engines.node` in various `package.json`s to require fewer adjustments in the future, by relying on a minimum Node.js version without an explicit maximum version. Lastly, it upgrades Nuxt 2 projects to use the latest Nuxt 2 version, for Node.js 18 compatibility.

## How to Test

1. Check out the `ENG-9038/node-18` branch
2. Ensure that Node.js version `18.x` is installed and active
3. `npm run clean` from the monorepo root
4. `npm i` from the monorepo root
5. Navigate to `reference-stores/next` and `packages/storefront-sdk`, then `npm run build`

## Checklist



[ENG-9038]: https://nacelle.atlassian.net/browse/ENG-9038?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
